src/inpage/QryptoAccount.ts

```diff
@@ -0,0 +1,61 @@
+const dedent = require('dedent-js');
+
+import { postWindowMessage } from '../utils/messenger';
+import { TARGET_NAME, API_TYPE } from '../constants';
+
+const INIT_VALUES = {
+  accountIsLoggedIn: false,
+  name: '',
+  network: '',
+  address: '',
+  balance: 0,
+};
+
+export class QryptoAccount {
+  public accountIsLoggedIn: boolean = INIT_VALUES.accountIsLoggedIn;
+  public name: string = INIT_VALUES.name;
+  public network: string = INIT_VALUES.network;
+  public address: string = INIT_VALUES.address;
+  public balance: number = INIT_VALUES.balance;
+
+  constructor() {
+    postWindowMessage(TARGET_NAME.CONTENTSCRIPT, {
+      type: API_TYPE.GET_INPAGE_QRYPTO_ACCOUNT_VALUES_1,
+      payload: {},
+    });
+  }
+
+  /* This method gives a tie in to the dapp, so that it can be notified
+  when a user logs in or out of qrypto. The dapp overwrites this method.
+  */
+  public statusChanged() {
+    const text = dedent`window.qrypto.account has changed.
+    Overwrite the function window.qrypto.account.statusChanged to make this notification interact with your dapp.
```

Comment:
not mandatory to override, but this allows them to get notified when qrypto has changed instead of needing to continually poll the qrypto.account object

Reply:
do you think having them continually poll is better?

Reply:
To clarify - I am calling this method whenever qrypto.account changes, so if the dapp creator overrides it, whatever function he overwrites it with will get called whenever qrypto.account changes. The alternative is for them to continually check if qrypto.account has changed.

Reply:
I was originally thinking of use window `messages` to alert the user as that is generally the standard to get alerted of something. window messages can be constantly listened on, but don't think `console.logs` can.

think it might confuse them that the other way to receive some data was via a `message`, but then this one is notified via a console/function call.

Reply:
-mm if the dapp developer overwrites the method with their function  own function, then there func will just get called when qrypto.account changes;  they don't need to listen to the console.log, the console.log would actually be overwritten at that point and no longer happen.
To further clarify:
1)<qrypto.account changes>
2) calls ==> statusChanged (the method statusChaned could actually be empty, it doesnt matter, the dapp developer implements/overwrites the function to do whatever he wants to do when qrypto.account has changed) 
3)Dapps statusChanged method calls a function that refetches qrypto.account.balance and displays it on the webpage 

-this is just something I came up with though, so happy to change it. Im not clear how to do it with window messages do you mean window.postMessage and let the dapp  listen for teh event and check for some type flag?(it's not really cross origin at this point so I wouldn't think you would need a window.postMessage? I would think you may as well just call the dapp function directly since this code is injected?)  or do you mean something else?

Reply:
What do you think about having a long-lived connection? https://developer.chrome.com/apps/messaging#connect

Could be a little overkill though.

While I agree that your method works, it just doesn't seem apparent or a standard way of doing things in the web environment. The primary method of data communication seems to be message passing.

Reply:
I researched this a bit, and I think using long-lived connections make sense. Long-lived connections trigger a Port.onDisconnect which we can use to help manage which webpages we need to send any account change notifications to.